src/rule_definition_tools.jl

```diff
@@ -258,3 +249,144 @@ This is able to deal with fairly complex expressions for `f`:
 propagator_name(f::Expr, propname::Symbol) = propagator_name(f.args[end], propname)
 propagator_name(fname::Symbol, propname::Symbol) = Symbol(fname, :_, propname)
 propagator_name(fname::QuoteNode, propname::Symbol) = propagator_name(fname.value, propname)
+
+"""
+    @non_differentiable(signature_expression)
+
+A helper to make it easier to declare that a method is not not differentiable.
+This is a short-hand for defining a [`frule`](@ref) and an [`rrule`](@ref)+ pullback that 
+returns [`DoesNotExist()`](@ref) for all partials (except for the function `s̄elf`-partial
+itself which is `NO_FIELDS`)
+
+The usage is to put the macro before a function signature.
+Keyword arguments should not be included.
```

Comment:
```suggestion
A helper to make it easier to declare that a method is not not differentiable.
This is a short-hand for defining an [`frule`](@ref) and [`rrule`](@ref) that 
return [`DoesNotExist()`](@ref) for all partials (except for the function `s̄elf`-partial
itself which is `NO_FIELDS`)

Keyword arguments should not be included.
```

Grammar + removed line about usage, because it wasn't clear from it what the correct usage was over-and-above what you would usually do with a macro, and the doctests provide far more information.